# Titanic Data Analysis
## Feature Engineering
- *Goal*: refine feature engineering (age binning and extract deck from cabin) to improve passenger prediction

In [8]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Import Dataframe
df = pd.read_csv('01_Titanic_Case_Study/Titanic_Train.csv')
print('--Data Load')

--Data Load


In [9]:
# General inspection of the Dataframe with head(), info() and insa().sum()
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [11]:
df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [12]:
# Filling missing values in Cabin with 'unknown' and extracting Deck
df['Cabin'] = df['Cabin'].fillna('Unknown')

# Extracting the first letter from the non-missing values
df['Deck'] = df['Cabin'].str[0]

# Checking the values
print(df['Deck'].unique())

['U' 'C' 'E' 'G' 'D' 'A' 'B' 'F' 'T']


**Note:** In the previous FE file the Cabin values were changed into hasCabin with 0 = False and 1 = True, now the first letter was extracted to improve the feature engineering for a better passenger prediction later. <br>
The U Deck is for 'unkown'

In [14]:
print(df.groupby('Deck')['Survived'].mean())

Deck
A    0.466667
B    0.744681
C    0.593220
D    0.757576
E    0.750000
F    0.615385
G    0.500000
T    0.000000
U    0.299854
Name: Survived, dtype: float64


**Note:** Best decks: B, D and E with a survival rate over 70%, except deck C which shows a survival rate of only 59%. The survival rate on deck F and G and the unknown decks are significant lower which indicates, that in these decks are more down in the ship and mostly used by passengers traveling in the third class. 